src/Server/Workspace.php

```diff
@@ -200,4 +214,44 @@ public function xdependencies(): array
         }
         return $dependencyReferences;
     }
+
+    /**
+     * @param Options|null $settings
+     */
+    public function didChangeConfiguration(Options $settings = null)
+    {
+        if ($settings === null) {
+            return;
+        }
+
+        $changedOptions = $this->getChangedOptions($settings);
+        $this->options = $settings;
```

Comment:
This should just do
```php
foreach (get_object_vars($settings) as $prop => $val) {
  $this->options->$prop = $val;
}
```
then the options will get updated.

Do you have any specific concerns over sharing the Options instance?